tka/sig.go

```diff
@@ -42,17 +48,47 @@ func (s SigKind) String() string {
 type NodeKeySignature struct {
 	// SigKind identifies the variety of signature.
 	SigKind SigKind `cbor:"1,keyasint"`
-	// Pubkey identifies the public key which is being certified.
+	// Pubkey identifies the public key which is being authorized.
 	Pubkey []byte `cbor:"2,keyasint"`
 
 	// KeyID identifies which key in the tailnet key authority should
 	// be used to verify this signature. Only set for SigDirect and
 	// SigCredential signature kinds.
 	KeyID []byte `cbor:"3,keyasint,omitempty"`
 
-	// Signature is the packed (R, S) ed25519 signature over the rest
-	// of the structure.
+	// Signature is the packed (R, S) ed25519 signature over all other
+	// fields of the structure.
 	Signature []byte `cbor:"4,keyasint,omitempty"`
+
+	// Nested describes a NodeKeySignature which authorizes the node-key
```

Comment:
I think this might be worth some elaboration.

I believe from the rest of the code that `Nested` contains the original `NodeKeySignature` that approved this node (an approved verification key) _or_ the previous rotation's `NodeKeySignature` (which will in turn nest more rotations, eventually terminating in the original approval). Is that correct?

I wonder if it would be better to make it an array since the nesting might get pretty deep and it'll be more efficient to allocate the entire array upfront.

It might also be good to have a plan for compaction even if we don't implement that right away.

Reply:
For compaction: My hope was that control would (eventually) implement telling a node with a network-lock key to direct-sign a signature, providing the non-direct signature as evidence that the node-key is authorized.

I'm hoping to avoid nesting more than 2 or three deep. The rotation key shouldnt change (or shouldnt change frequently), so when the rotation happens, the node can drop the outer signature, and make a new one using its rotation key. IE: it stays 2 deep even when it rotates.

> I believe from the rest of the code that Nested contains the original NodeKeySignature that approved this node (an approved verification key) or the previous rotation's NodeKeySignature (which will in turn nest more rotations, eventually terminating in the original approval). Is that correct?

Yes, with the caveat that I don't want the nesting to grow unbounded as per above.

Reply:
Okay, I guess I was expecting that the rotation key would also rotate at each rotation. Maybe add a comment about that too!

Reply:
Done.